File: Task-03/Elixir/Sub-Task3.exs
def main do
    IO.puts "Enter a number:"
    n = String.to_integer(IO.gets("") |> String.trim)
    print_diamond(n)
  end
  defp print_diamond(n) do
    Enum.each(1..n, fn i -> print_line(i, n) end)
    Enum.each((n-1)..1, fn i -> print_line(i, n) end)
  end

  defp print_line(i, n) do
    spaces = n - i
    stars = 2 * i - 1

    IO.puts String.duplicate(" ", spaces) <> String.duplicate("*", stars)
  end
end

DiamondPattern.main()

File: Task-03/Ruby/Sub-Task3.rb
def print_diamond(n)
(0...n).each do |i|
    puts " " * (n - i - 1) + "* " * (i + 1)
  end
(n - 2).downto(0) do |i|
    puts " " * (n - i - 1) + "* " * (i + 1)
  end
end
print "Enter the number of rows for the diamond pattern: "
n = gets.to_i
print_diamond(n)
